scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -44,7 +44,20 @@ object TypedPipe extends Serializable {
    */
   def from[T](pipe: Pipe, fields: Fields)(implicit flowDef: FlowDef, mode: Mode, conv: TupleConverter[T]): TypedPipe[T] = {
```

Comment:
I do think so, the only question is when to do that. I think we can basically pull all the cascading specific stuff into an enrichment now and stay source compatible. I think your plan is exactly what I want to drive towards: scalding-typed is a dependency of scalding-core and does not depend on cascading at all.